internal/rbd/controllerserver.go

```diff
@@ -786,38 +828,9 @@ func (cs *ControllerServer) DeleteVolume(
 	rbdVol, err := genVolFromVolID(ctx, volumeID, cr, req.GetSecrets())
 	defer rbdVol.Destroy()
 	if err != nil {
-		if errors.Is(err, util.ErrPoolNotFound) {
-			log.WarningLog(ctx, "failed to get backend volume for %s: %v", volumeID, err)
-
-			return &csi.DeleteVolumeResponse{}, nil
-		}
-
-		// if error is ErrKeyNotFound, then a previous attempt at deletion was complete
-		// or partially complete (image and imageOMap are garbage collected already), hence return
-		// success as deletion is complete
-		if errors.Is(err, util.ErrKeyNotFound) {
-			log.WarningLog(ctx, "Failed to volume options for %s: %v", volumeID, err)
-
-			return &csi.DeleteVolumeResponse{}, nil
-		}
-
-		// All errors other than ErrImageNotFound should return an error back to the caller
-		if !errors.Is(err, ErrImageNotFound) {
-			return nil, status.Error(codes.Internal, err.Error())
-		}
-
-		// If error is ErrImageNotFound then we failed to find the image, but found the imageOMap
-		// to lead us to the image, hence the imageOMap needs to be garbage collected, by calling
-		// unreserve for the same
-		if acquired := cs.VolumeLocks.TryAcquire(rbdVol.RequestName); !acquired {
-			log.ErrorLog(ctx, util.VolumeOperationAlreadyExistsFmt, rbdVol.RequestName)
-
-			return nil, status.Errorf(codes.Aborted, util.VolumeOperationAlreadyExistsFmt, rbdVol.RequestName)
-		}
-		defer cs.VolumeLocks.Release(rbdVol.RequestName)
-
-		if err = undoVolReservation(ctx, rbdVol, cr); err != nil {
-			return nil, status.Error(codes.Internal, err.Error())
+		delRes, err2 := cs.checkErrAndUndoReserve(ctx, err, volumeID, rbdVol, cr)
+		if err2 != nil {
+			return delRes, err2
```

Comment:
no need to check the `err2` here directly return it
and also I would suggest changes as below
```
return cs.checkErrAndUndoReserve(ctx, volumeID, rbdVol, cr,err)
```
or 
```
var resp = *csi.DeleteVolumeResponse
resp,err=cs.checkErrAndUndoReserve(ctx, volumeID, rbdVol, cr,err)
return resp,err
```

commit title can be updated from `rbd: add a new func() for error check and unreserve omap in delete` to  ` rbd: add checkErrAndUndoReserve for error check and unreserve omap in deleteVolume`

Reply:
yeah, return cs.checkErrAndUndoReserve(ctx, volumeID, rbdVol, cr,err) should be ok, if there are some changes happening in checkErrAndUndoReserve it helps.